lib/matplotlib/collections.py

```diff
@@ -1827,12 +1828,20 @@ def determine_facecolor(patch):
                 if patch.get_fill():
                     return patch.get_facecolor()
                 return [0, 0, 0, 0]
-
+            def is_default(c):
+                return mcolors.same_color([0, 0, 0, 0], c)
             kwargs['facecolors'] = [determine_facecolor(p) for p in patches]
-            kwargs['edgecolors'] = [p.get_edgecolor() for p in patches]
+            edgecolors = [p.get_edgecolor() for p in patches]
+            # Since the hatch color is updated when the edge color
+            # is updated, we will ignore the edge color if it is blank (the
+            # default of [0, 0, 0, 0]).
+            if not all(is_default(c) for c in edgecolors):
+                kwargs['edgecolors'] = edgecolors
             kwargs['linewidths'] = [p.get_linewidth() for p in patches]
             kwargs['linestyles'] = [p.get_linestyle() for p in patches]
             kwargs['antialiaseds'] = [p.get_antialiased() for p in patches]
+            if patches:
+                kwargs['hatch'] = patches[0].get_hatch()
```

Comment:
If we are going to only pick of the hatch from the first patch, then we should warn or raise if we are going to ignore _the rest_ of the hatches.

Reply:
Why? In that's true, it would be impossible to make a collection of multiple patches w/ hatches without getting a warning.

Reply:
sorry I was not clear.  The check should be "are all of the hatches the same? if so carry on, if not fail!"  As implemented if the user passes in 

```python
[
        mpatches.Rectangle((0, 0), 0.2, .2, hatch="/"),
        mpatches.Rectangle((0.4, 0.4), 0.2, .2, hatch="\\")
]
```

then the first hatch will be used for both and we will silently drop the `"\\"` on the second one!